### PR TITLE
[ET-221] Move SAML/LDAP check into pre-flight

### PR DESCRIPTION
**Pull request by chefsalim**
This change moves the check for SAML/LDAP into the pre-flight checks.
It adds a new AuthPreflightValidator class.

### CODE REVIEW

File: omnibus/files/private-chef-cookbooks/private-chef/libraries/preflight_auth_validator.rb [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+  private
+
+  def validate_sane_state
+    # used for checking if LDAP and SAML are both enabled
+    chef_manage = { "chef_manage" => {} }
+    if ::File.exists?("/var/opt/chef-manage/etc/chef-manage-running.json")
+      chef_manage = JSON.parse(IO.read("/var/opt/chef-manage/etc/chef-manage-running.json"))
+    end
+    node.consume_attributes({"chef_manage" => chef_manage["chef-manage"]})
```
Comment (stevendanna) [inline omnibus/files/private-chef-cookbooks/private-chef/libraries/preflight_auth_validator.rb]: If we only need this in the check, why merge this into the node object at all?

**Comment by marcparadise** [comment]
:+1: pending ssd's suggestion to just pull the  attributes we need into the node, since there's just 1

**Comment by chefsalim** [comment]
Updated w/ change to just pull the attributes instead of merging.

**Comment by stevendanna** [comment]
:+1:

**Pull request merged**